strongbox-event-api/src/main/java/org/carlspring/strongbox/event/EventExecutorFactoryBean.java

```diff
@@ -0,0 +1,130 @@
+package org.carlspring.strongbox.event;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+
+import javax.servlet.ServletContext;
+
+import org.apache.catalina.Service;
+import org.apache.catalina.core.ApplicationContext;
+import org.apache.catalina.core.ApplicationContextFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.core.task.SyncTaskExecutor;
+import org.springframework.data.util.ReflectionUtils;
+
+public class EventExecutorFactoryBean implements FactoryBean<Executor>
+{
+
+    private static final String STRONGBOX_HTTP_REQUEST_EXECUTOR = "strongbox.httpRequestExecutor";
+
+    private static final Logger logger = LoggerFactory.getLogger(EventExecutorFactoryBean.class);
+
+    private final ServletContext servletContext;
+
+    public EventExecutorFactoryBean(ServletContext servletContext)
+    {
+        super();
+        this.servletContext = servletContext;
+    }
+
+    @Override
+    public Executor getObject()
+        throws Exception
+    {
+        Executor executor = Optional.ofNullable(servletContext)
+                                    .flatMap(c -> Optional.ofNullable(lookupExecutor()))
+                                    .orElse(new SyncTaskExecutor());
+        
+        logger.info(String.format("Using [%s] executor for Async events.", executor.getClass()));
+        
+        return executor;
+    }
+
+    private Executor lookupExecutor()
+    {
+        Executor result;
+        if ((result = lookupJettyExecutor()) != null)
+        {
+            return result;
+        }
+        else if ((result = lookupTomcatExecutor()) != null)
+        {
+            return result;
+        }
+        return null;
+    }
+
+    private Executor lookupJettyExecutor()
+    {
+        Executor executor = (Executor) servletContext.getAttribute("org.eclipse.jetty.server.Executor");
+        if (executor == null)
+        {
+            return null;
+        }
+
+        logger.info("Jetty environment detected.");
+
+        return executor;
+    }
+
+    private Executor lookupTomcatExecutor()
+    {
+        try
+        {
+            Class.forName("org.apache.catalina.core.ApplicationContextFacade");
```

Comment:
Isn't it possible to get the `executor` from the `servletContext` in a similar way to how you're doing it in the `lookupJettyExecutor` method?

Reply:
seems no, at least I was not able to find alternatives